tests/testsuite/features2.rs

```diff
@@ -1406,6 +1406,41 @@ workspace: [..]/foo/Cargo.toml
         .run();
 }
 
+#[cargo_test]
+fn edition_2021_workspace_member() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [workspace]
+            members = ["a"]
+            "#,
+        )
+        .file(
+            "a/Cargo.toml",
+            r#"
+            [package]
+            name = "a"
+            version = "0.1.0"
+            edition = "2021"
+            "#,
+        )
+        .file("a/src/lib.rs", "")
+        .build();
+
+    p.cargo("check")
+        .with_stderr(
+            "\
+warning: some crates are on edition 2021 which defaults to `resolver = \"2\"`,
+    but a virtual workspace defaults to `resolver = \"1\"`
+    specify the desired resolver version explicitly in the workspace root's manifest
```

Comment:
nit: this feels like we are commanding the user and not really giving details.

Maybe: "to use the 2021 edition resolver, specify `workspace.resolver = "1"`"